frontend/src/container/NewDashboard/DashboardVariablesSelection/DashboardVariableSelection.tsx

```diff
@@ -43,35 +53,26 @@ function DashboardVariableSelection(): JSX.Element | null {
 		}
 	}, [variables]);
 
-	const onVarChanged = (name: string): void => {
-		/**
-		 * this function takes care of adding the dependent variables to current update queue and removing
-		 * the updated variable name from the queue
-		 */
-		const dependentVariables = variablesTableData
-			?.map((variable: any) => {
-				if (variable.type === 'QUERY') {
-					const re = new RegExp(`\\{\\{\\s*?\\.${name}\\s*?\\}\\}`); // regex for `{{.var}}`
-					const queryValue = variable.queryValue || '';
-					const dependVarReMatch = queryValue.match(re);
-					if (dependVarReMatch !== null && dependVarReMatch.length > 0) {
-						return variable.name;
-					}
-				}
-				return null;
-			})
-			.filter((val: string | null) => !isNull(val));
-		setVariablesToGetUpdated((prev) => [
-			...prev.filter((v) => v !== name),
-			...dependentVariables,
-		]);
-	};
+	const initializationRef = useRef(false);
+
+	useEffect(() => {
+		if (variablesTableData.length > 0 && !initializationRef.current) {
```

Comment:
why is this required ? 

- since the initial state of dependency data is null, a null check would be sufficient to know the first load ? 
- on similar lines we do not want to re-calculate the deps graph / order when `variablesTableData` is updated  ? when adding or removing variables from the dashboard ?

Reply:
done, removed ref and added a check on the dependencyGraph as discussed

Reply:
Also tested variable addition and deletion